PR: 6.8.8

edreamleo opened the pull request:
This PR summarizes all changes made in Leo 6.8.8. It will remain a draft until Leo 6.8.8 goes out the door.

This PR will not affect LeoJS in any way, except possibly for the script helpers in `leo/commands/editFileCommands.py`.

<details><summary><b>Leo 6.8.8: All significant changes</b></summary>
<br>

Here is a high-level summary of all significant changes made in Leo 6.8.8, not including formatting changes made by ruff format.

This script helped make sense of all the changes:
```python
g.cls()
from leo.commands import editFileCommands as efc
# Any branch or rev is valid.
efc.GitDiffController(c).summary_diff_two_revs(
    rev1='master',    # Old
    rev2='6.8.8',     # New
    path='leo/core',  # Limit diffs to this directory, relative to the nearest git repo.
)  # fmt: skip
```

Use `cff # fmt: skip` to see where I retained Leo's legacy formatting. These typically involve columnar tables.

**[PR #4477](https://github.com/leo-editor/leo-editor/pull/4477): leo/core/leoColorizer.py**

- Fix a colorizer bug involving adjacent `@doc` parts.

**[PR #4484](https://github.com/leo-editor/leo-editor/pull/4484): leo/scripts/check_leo.py**

- Add `leo/scripts/check_leo.py`.
- Make minor, non-trivial changes to other files. See the PR for details.

**[PR #4486](https://github.com/leo-editor/leo-editor/pull/4486): leo/commands/commanderFileCommands.py**

- Improve `flatten-outline-to-node`.

**[PR #4490](https://github.com/leo-editor/leo-editor/pull/4490) and [PR #4518](https://github.com/leo-editor/leo-editor/pull/4518): All files**

Retire Leo's beautify commands and beautify all of Leo's files with ruff format using ruff's default line length of 100.
The more I use this line length the better I like it.

*Important*: PR #4490 and several later PRs (not listed here) changed the formatting of *all* of Leo's source files.
*This* PR (PR #4487) shows the final results of all formatting!
I have carefully reviewed all the diffs for the `leo/core` and `leo/commands` directories, as well as Leo's Qt gui files in `leo/plugins`.

Significant changes to files:

- `leoApp.py`: Remove command-line arguments.
- `leoAst.py`: Move classes related to legacy formatter to the attic.
- `leoAtFile.py`: Add support for ruff format.
- 'leoTokens.py`: Delete the file, moving its contents to the attic.

Features:

- Add `ruff format` settings to `pyproject.toml`. Specify default line length of 100 (in a later PR).
- Remove `-b` and `--black-sentinels` command-line arguments.
- Add `@bool write-black-sentinals` setting, with default `False`.
- Replace `g.app.write_black_sentinels` with `c.write_black_sentinel`.
- Rewrite `leo/scripts/beautify_all_leo.py` and delete `leo/scripts/beautify_leo.py`
- Replace all `@beautify` and `@nobeautify` directives with the `# fmt: on`, `# fmt: off`, and `# fmt: skip` pragma comments.
- Beautify all of Leo's core files with `ruff format`.
- Remove some flake8 suppressions.

**[PR 4498](https://github.com/leo-editor/leo-editor/pull/4498): leo/core/leoExternalFiles.py, leo/core/leoExternalFiles.py**

- Updating external files no longer changes `c.p`.

**[PR #4502](https://github.com/leo-editor/leo-editor/pull/4502), [PR #4513](https://github.com/leo-editor/leo-editor/pull/4513), and [PR #4517](https://github.com/leo-editor/leo-editor/pull/4517): leo/commands/editFileCommands.py**

- Add `summary_diff_two_revs` and its helpers.
- Add `path` kwarg to limit scope of diffs.
- Show the `path` kwarg in the top-level node if it exists.
- Improve status message for long diffs.

**[PR #4510](https://github.com/leo-editor/leo-editor/pull/4516): leo/commands/abbrevCommands.py**

- Fix bugs involving abbreviations.

</details>

tbpassin reviewed (commented):
I don't think we need to include items like:
> [Merge branch 'devel' into ekr-4462-date-abbrev](https://github.com/leo-editor/leo-editor/pull/4487/commits/393e679b1d59d288eed6659e3cb87db0e0512e8a)

That merge isn't something that affected 6.8.8.

edreamleo commented:
@tbpassin Thanks for your comment.

This PR isn't about what you or I think *should* be included in 6.8.8&mdash;it records what *already has* been included. In other words, everything here is the result of the diffs between the `master` and `6.8.8` branches.

If something isn't correct, we would have to undo that with another PR that becomes merged with the `6.8.8` branch. Clear?

tbpassin commented:
That's clear but I don't think that merging devel into a branch that may become a PR should count.  When that PR gets merged back into devel, of course that's what we want to note.

tbpassin commented:
I'm not sure what the status of all the changes is. But reviewing hundreds of diffs that are caused by beautification event with the new summarizing-diffs script, will be quite a challenge. If you have not done so already, may I suggest creating another branch and have it contain only the parts that contain beautification code. That branch would be quite reviewable. Once that branch gets merged, then we could deal with the changes induced by running a beautification pass.

Of course, this may already have been accomplished. It's hard for me to tell.

edreamleo commented:
@tbpassin About the status of this PR (and Leo 6.8.8):

- I have completed all the code cleanups.
- I have just filed #4505. I'll get to this asap.

Thomas, I don't think you need to worry about the huge recent diffs. Instead, I suggest you concentrate only on your VR3 plugin, and any other plugins that are important to you.

I urge you to trust both me and `ruff format` to have done the right things.  Trust, as in "trust/verify" :-) I'll deal with any new bugs as people find them.

But if you really want to check recent work, the place to start would be to do a `cff` on `# fmt:`. That search will reveal almost all the "by hand" work I have done recently.

Finally, I'll soon fully document all the recent PRs in `leoDocs.leo`, paying particular attention to:
- New user options.
- Features that might impact LeoJS.

edreamleo commented:
@tbpassin My improved summary diff for VR3 shows that two tuples, `RST_ERROR_BODY_STYLE` and `RST_ERROR_MSG_STYLE`, have been squished. If you like, I can create a PR that undoes those changes with two `# fmt: skip` statements. Would you like me to do that?

tbpassin commented:
> @tbpassin My improved summary diff for VR3 shows that two tuples, `RST_ERROR_BODY_STYLE` and `RST_ERROR_MSG_STYLE`, have been squished. If you like, I can create a PR that undoes those changes with two `# fmt: skip` statements. Would you like me to do that?

Yes, please do.

edreamleo commented:
@tbpassin I'm on it :-)

edreamleo commented:
@tbpassin Rev [f57435f](https://github.com/leo-editor/leo-editor/pull/4487/commits/f57435fd67e808e19300f8d5ed231bad87176da7) restores the formatting.

edreamleo committed:
Use the PR for 6.8.8 to describe all the ruff format changes

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Add path kwarg to limit scope of diffs. Remove @ignore from summary diffs

edreamleo committed:
Merge pull request #4513 from leo-editor/ekr-limited-git-diffs

PR: improve git diff helpers

edreamleo committed:
Update LeoPyRef.leo

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Reformat cmd decorators by removing 'magic' commas

edreamleo committed:
Merge pull request #4514 from leo-editor/ekr-smaller-at-cmd

PR: Reformat cmd decorators by removing 'magic' commas

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Simplify more calls to g.new_cmd_decorator

edreamleo committed:
Merge pull request #4515 from leo-editor/ekr-smaller-at-cmd2

PR2: Reformat cmd decorators by removing 'magic' comma.

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
PR fixes #4510: ',,' in link-pr

edreamleo committed:
Merge pull request #4516 from leo-editor/ekr-4510-abbrev

PR fixes #4510: comma-comma expansion

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
PR: improve git diff script helpers

edreamleo committed:
Merge pull request #4517 from leo-editor/ekr-report-git-path-kwarg

PR: improve git diff script helpers

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Merge branch 'ekr-line-length-88' into ekr-line-length-100

edreamleo committed:
Reformat all files with line-length = 100

edreamleo committed:
Fix nits by adding fmt: skip or by moving comments

edreamleo committed:
Reformat leoApp.py

edreamleo committed:
Beautify leoApp.py

edreamleo committed:
Tweak five core files

edreamleo committed:
Tweak leoCommands.py

edreamleo committed:
Tweak leoFrame.py

edreamleo committed:
Tweak four more files

edreamleo committed:
Suppress a line-too-long message

edreamleo committed:
Merge pull request #4518 from leo-editor/ekr-line-length-100

PR2: beautify all of Leo's files with line-length = 100

edreamleo committed:
Reformat anki.py

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Update LeoPyRef, setting @int black-line-length = 100

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
PR: tweaks after changing black-line-length to 100

edreamleo committed:
Merge pull request #4519 from leo-editor/ekr-format-tweaks-1-17-2026

PR: tweaks after changing black-line-length to 100

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Remove magic comma

edreamleo committed:
Remove ancient comments containing % as a string op

edreamleo committed:
Fix a minor bug in p.badUnlink

edreamleo committed:
Convert to f-strings in a few places

edreamleo committed:
Convert two more % operators. Simplify 'line-number' command

edreamleo committed:
Merge pull request #4520 from leo-editor/ekr-cleanups-1-17-2026

PR: yet more minor code cleanups

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Update release notes

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Improve 'line-number' docstring

edreamleo committed:
Update status var

edreamleo committed:
Fix #4422, retiring several methods

edreamleo committed:
Retire c.redraw_after_expand. Make c.redraw_after_expand/contract synonyms for c.redraw

edreamleo committed:
c.redraw_after_contract and c.redraw_after_expand synonyms for c.redraw

edreamleo committed:
tweak traces

edreamleo committed:
Tweak traces

edreamleo committed:
qtree.onItemCollapsed and qtree.onItemExpanded always call c.redraw

edreamleo committed:
Improve docstrings

edreamleo committed:
Renamed a slow unit test

edreamleo committed:
Only trace 3 callers

edreamleo committed:
Simplify lockouts

edreamleo committed:
c.expandToLevel always calls c.redraw

edreamleo committed:
Merge pull request #4523 from leo-editor/ekr-4522-contract-parent

PR Fixes #4522: contract-parent

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Two more improvements

edreamleo committed:
Remove duplicate commands from commanderEditCommands.py

edreamleo committed:
Fix crasher

edreamleo committed:
Merge pull request #4527 from leo-editor/ekr-4524-exchange-point-mark

PR fixes #4524: Fix crasher in qtew.leoMoveCursorHelper

edreamleo committed:
Merge pull request #4521 from leo-editor/ekr-cleanups-1-26-2006

Fix two more quirps

edreamleo committed:
Merge pull request #4526 from leo-editor/ekr-4525-duplicate-code

PR: Remove duplicate goto commands from commanderEditCommands.py

edreamleo committed:
Avoid using recently-retired names

edreamleo committed:
Document a minor breaking API change

edreamleo committed:
Merge branch 'devel' into 6.8.8

boltex committed:
Escape HTML entities BEFORE converting URLs to clickable links!

edreamleo committed:
Merge pull request #4528 from leo-editor/felix-fix-html-export

Escape HTML entities BEFORE converting URLs to clickable links!

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Fix #4528

edreamleo committed:
Add BaseColorizer.last_trace and associated logic

edreamleo committed:
Improve cached traces: add state info

edreamleo committed:
Don't replace entire headline

edreamleo committed:
Remove unnecessary guard

edreamleo committed:
Restore selection and continue editing

edreamleo committed:
Merge pull request #4530 from leo-editor/ekr-4529-simple-headline-abbreviations

PR fixes #4528: simple abbreviations in headlines

edreamleo committed:
Merge branch 'devel' into ekr-4398-colorizer

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Improve dumps

edreamleo committed:
Fix a new pylint complaint.

edreamleo committed:
Remove dead code

edreamleo committed:
Merge pull request #4531 from leo-editor/ekr-4398-colorizer

PR: Improve colorizer traces

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Make PygmentsColorizer a subclass of JEditColorizer

edreamleo committed:
Move init_style_ivars into the JEditColorizer class

edreamleo committed:
Print a warning if pygments has not been installed

boltex committed:
Ensured proper tree re-rendering on separator drag.

wangzhaohe committed:
fix(asciidoctor): Fix not get rvm dir & without ignore head/node/tree

wangzhaohe committed:
feat(plugin): 1st body_style_hook.py

wangzhaohe committed:
feat(plugin): 2nd body_style_hook.py can run well but multiple .leo

wangzhaohe committed:
feat(plugin): 3rd body_style_hook.py open new .leo and tab-detach both ok

wangzhaohe committed:
feat(plugin): 4th body_style_hook.py Use English comments

wangzhaohe committed:
feat(plugin): 5th body_style_hook.py save as leo @file format

boltex committed:
Fixed again the rendertree decision when dragging separators.
Needs both ways because it may have to set widths of selections in outline horizontally, while having to render more rows vertically.

wangzhaohe committed:
feat(adoc): Add g.getScript() to expand << section Reference >>

wangzhaohe committed:
feat(adoc): Solve adoc expand @others rendered below again

edreamleo committed:
Remove redundant call

edreamleo committed:
Merge pull request #4539 from leo-editor/ekr-readAtFileNodes-quirp

PR: Remove redundant call to at.readAtFileNode

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Merge pull request #4533 from leo-editor/ekr-simpler-colorizer-classes

PR: Simplify the classes in leoColorizer.py

edreamleo committed:
Merge branch 'devel' into 6.8.8

wangzhaohe committed:
Fix Cmd+shift+( or ), abbrevation shortkey、shift-tab cause body style missing bug

wangzhaohe committed:
fix: Fix Cmd+shift+( or ) / shift+tab flashing bug

wangzhaohe committed:
fix: Fix leo abbrivation flash bug

wangzhaohe committed:
feat: Add @data body-style-settings in myLeoSettings.leo

wangzhaohe committed:
chore: Rename body_style_hook.py to body_style.py

bkmgit committed:
Add metadata files for desktop GNU/Linux packaging

wangzhaohe committed:
fix: Restore missing sequence check in JEditColorizer.match_eol_span

wangzhaohe committed:
chore(body_style): Change #@ to # @

edreamleo committed:
Merge pull request #4550 from wangzhaohe/fix-color-final

fix: Restore missing sequence check in JEditColorizer.match_eol_span

edreamleo committed:
Remove @bool write-black-sentinels and related ivar

edreamleo committed:
Remove @bool write-black-sentinels from two files

edreamleo committed:
Merge branch 'devel' into 6.8.8

wangzhaohe committed:
fix(adoc): << section >> body text double include in .adoc file

edreamleo committed:
Update documentation

edreamleo committed:
Merge pull request #4548 from bkmgit/linux-packaging

Add metadata files for desktop GNU/Linux packaging

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Merge pull request #4551 from leo-editor/ekr-4545-write-black-sentinels

Remove @bool write-black-sentinels and related ivar

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Add do-nothing md_i.postprocess and corresponding unit test

wangzhaohe committed:
fix(body-style): Issue: Scroll Jump and UI Flickering on Undo in Body Pane

wangzhaohe committed:
fix(body_style): Issue: Cursor Resets to Top of Pane during Redo (Ctrl+Shift+Z)

wangzhaohe committed:
fix(body_style): Fixed Compatibility with Native Zoom (zoom-in / zoom-out)

edreamleo committed:
Simplify mdw.write as suggested by Félix

gatesphere committed:
Fix quicksearch.py body matches for Python 3.14

edreamleo committed:
Merge pull request #4553 from gatesphere/quicksearch-fix-py3.14

Fix quicksearch.py body matches for Python 3.14

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Merge pull request #4552 from leo-editor/ekr-markdown-blank-lines

Ensure blank lines between imported markdown sections

edreamleo committed:
Merge pull request #4534 from leo-editor/felix-fix-html-export

Ensured proper tree re-rendering on separator drag.

edreamleo committed:
Merge branch 'devel' into 6.8.8

wangzhaohe committed:
fix: Change Iconbar button hover text color to orange in SwotMoknokai theme

edreamleo committed:
Fix typo (#4555)

edreamleo committed:
Merge pull request #4557 from wangzhaohe/SwotMoknokai

fix: Change Iconbar button hover text color to orange in SwotMoknokai…

wangzhaohe committed:
fix(body_style): Fixed static analysis errors and unit test failures

edreamleo committed:
Merge pull request #4537 from wangzhaohe/body_style_hook_2026_02_19

Implementation of body_style_plugin for Consistent Line-Spacing and Typography #4536

edreamleo committed:
Reformat with black

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Make sure pyproject exists

edreamleo committed:
Remove dead code

edreamleo committed:
Sheesh: restore legacy quotes

edreamleo committed:
Restore more single quotes

boltex committed:
Fixed positioning of settings icon when resizing panes and improved colors of 'mark' related buttons.

edreamleo committed:
Merge pull request #4560 from leo-editor/ekr-4559-consistent-formatting

PR fixes #4559: consistent formatting options

edreamleo committed:
Merge pull request #4561 from leo-editor/felix-fix-html-viewer-colors-and-resize

Fixed positioning of settings icon when resizing panes and improved colors of 'mark' related buttons.

edreamleo committed:
Merge branch 'devel' into 6.8.8

bkmgit committed:
Remove unneeded dependency

tk is a machine learning library not the tkinter
framework which is part of base python.

boltex committed:
Made the set_name_and_title helper method also set tooltip on frame's tab. (setTabName now accepts optional tooltip)
Fixes #4558

edreamleo committed:
Merge pull request #4563 from bkmgit/tk

Remove unneeded dependency

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Rewrite setAllAncestorAtFileNodesDirty. Update dirty bits in efc.idle_check_commander

edreamleo committed:
Call the helper function!

edreamleo committed:
Second rewrite

edreamleo committed:
Revise v.setAllAncestorAtFileNodesDirty. Call it from efc.idle_check_commander

edreamleo committed:
Simplify

edreamleo committed:
Add the hidden vnode to seen

edreamleo committed:
Add to_do kwarg to v.setAllAncestorAtFileNodesDirty. efc.idle_check_commander calls it once

edreamleo committed:
Cleanups

edreamleo committed:
Rename the kwarg to to_do_set

edreamleo committed:
Revise docstring

edreamleo committed:
Hurray! No need for the result list

edreamleo committed:
Maintain compatibility

edreamleo committed:
Greatly simplify c.deletePositionsInList

edreamleo committed:
Cleanups

edreamleo committed:
A milestone. Make c.deletePositionsInList undoable!!

edreamleo committed:
Cleanups: remove obsolete theory of operation

edreamleo committed:
More cleanups

edreamleo committed:
Improve undo type

edreamleo committed:
Move items to the temp attic

edreamleo committed:
Eliminate redundant starting point

boltex committed:
Improved 'extract' to restrict finding defs  for JavaScript and TypeScript.
Left comment/note for further improvements of the extract command.
Fixes #4568

boltex committed:
Cleaned up comments

edreamleo committed:
Fix a typo

edreamleo committed:
Improve comments

edreamleo committed:
Merge pull request #4567 from leo-editor/ekr-simplify-delete-positions

PR: Greatly simplify c.deletePositionsInList

edreamleo committed:
Move the legacy theory of operation to the permanent attic

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Merge branch 'devel' into ekr-4565-update-at-clean-files

edreamleo committed:
Nested external files are no longer valid

edreamleo committed:
Mark all clones in the changed root's subtree as dirty

edreamleo committed:
Merge pull request #4569 from leo-editor/boltex/issue4568

Improved 'extract' to restrict finding defs for JavaScript and TypeScript.

edreamleo committed:
Merge branch 'devel' into ekr-4565-update-at-clean-files

edreamleo committed:
Consistency

boltex committed:
Fixed 'setTabName': no need to indicate that kwargs can be None.

edreamleo committed:
Merge pull request #4564 from leo-editor/boltex/issue4558

Made the set_name_and_title helper method also set tooltip on frame's tab. (setTabName now accepts optional tooltip)

edreamleo committed:
Merge pull request #4566 from leo-editor/ekr-4565-update-at-clean-files

PR fixes #4565: update @clean files

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
efc.ask does nothing unless a new setting is True. Give update messages in only one place

edreamleo committed:
Redraw only if changed!!!

edreamleo committed:
Simplify

edreamleo committed:
Add setting. Fix recent botch

edreamleo committed:
Remove unnecessary injected ivars

edreamleo committed:
Remove leo_geom_set ivar

edreamleo committed:
Fix #4574, revise unit test, remove @bool indent-added-comments

edreamleo committed:
Remove redundant headline comments

edreamleo committed:
Fix #4577. Update requirements

edreamleo committed:
Add traces

edreamleo committed:
Remove traces

edreamleo committed:
Merge pull request #4571 from leo-editor/ekr-4570-update-without-dialogs

PR fixes #4570: Add setting to bypass file-update dialogs.

edreamleo committed:
Merge pull request #4573 from leo-editor/ekr-remove-injected-wrappers

Remove ivars unnecessarily injected into gui classes

edreamleo committed:
Merge pull request #4575 from leo-editor/ekr-4574-python-comments

PR fixes #4574: preserve indentation when adding comments

edreamleo committed:
Merge pull request #4578 from leo-editor/ekr-4577-console-gui

PR fixes #4577

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Improve formatting

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Merge pull request #4535 from wangzhaohe/adoc-skip-ignore-node

Title: Core: Improve AsciiDoc path detection and @ignore-node level logic (#4424)

edreamleo committed:
Reformat one file

edreamleo committed:
Remove an error message that pollutes Leo's testing scripts

edreamleo committed:
Merge branch 'devel' into 6.8.8

boltex committed:
Fixed file type detection in 'ask' and updated the leoserver version of 'ask'.

boltex committed:
Cleaned up

boltex committed:
support flexible leoServerConfig in 'is_enabled' for the ExternalFilesControler class.

edreamleo committed:
Merge pull request #4579 from leo-editor/felix-fix-PR-4570

Fixed file type detection in 'ask' and updated the leoserver version of 'ask'.

edreamleo committed:
Beautify leoserver.py

edreamleo committed:
Merge branch 'devel' into 6.8.8

boltex committed:
Add EFC to the reloadConfigurableSettings table  and implement reloadSettings in EFC.

boltex committed:
Fixed stranded clone in LeoPyRef :)

boltex committed:
Reverted leopyref :)

edreamleo committed:
Merge pull request #4582 from leo-editor/boltex/issue4580

Boltex/issue4580

edreamleo committed:
Beautify after pulling PR

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Restore args to run_test_leo.py

edreamleo committed:
Tweak options code for unit tests

edreamleo committed:
Merge pull request #4584 from leo-editor/ekr-4-3-2026-fix-run_test_leo.py

PR: Restore args to run_test_leo.py

edreamleo committed:
Merge branch 'devel' into 6.8.8

boltex committed:
Made the setup of doShowMarked uniform with the other quicksearch commands.
Fixes #4585

edreamleo committed:
Merge pull request #4586 from leo-editor/boltex/issue4585

Made doShowMarked uniform with the other quicksearch commands.

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Run mypy w/o caching

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Update LeoPyRef.leo

edreamleo committed:
Update release files

edreamleo committed:
Fix markup error in appendix

edreamleo committed:
Update version in leoVersion.py and install scripts

edreamleo committed:
Remove broken link

edreamleo committed:
Change requirements.txt

edreamleo committed:
Update 6.8.8 date

boltex committed:
Added a leo-web entry below leojs on the front page. Also added a line about it at the end of Leo's history.

boltex committed:
Added LeoWeb in list of authored software.

boltex committed:
Fix string concatenation in abbreviation toggle message

edreamleo committed:
Merge pull request #4588 from leo-editor/felix-fix-toggle-abbrev-mode-command

Fix string concatenation in abbreviation toggle message

edreamleo committed:
Merge pull request #4589 from leo-editor/felix-add-leoweb-to-docs

Add LeoWeb references to the documentation.

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Fix announcement links

edreamleo committed:
Merge branch 'devel' into 6.8.8

edreamleo committed:
Update static date

edreamleo committed:
Fix a build error

edreamleo committed:
Remove trailing semicolon
